### PR TITLE
Fix crash bug when deleting a node with note pinned will crash Dynamo

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

Fix a crash bug when deleting a node with note pinned will crash Dynamo
(Will attach a snapshot soon)

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs [hunk 1]
```diff
@@ -344,10 +344,13 @@ private void SubscribeToPinnedNode()
             Model.PinnedNode.PropertyChanged += PinnedNodeModel_PropertyChanged;
             PinnedNode.PropertyChanged += PinnedNodeViewModel_PropertyChanged;
 
-            // Subscribe to pinnedNode.RequestSelection (fires before node is selected)
+            // Subscribe to PinnedNode.Selected (fires before node is selected)
             // so that this note is added to the selection
             PinnedNode.Selected += PinnedNodeViewModel_OnPinnedNodeSelected;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs:349]: Maybe outside of this task but can you check if this is unsubscribed correctly?

**Review by QilongTang — COMMENTED** [review]
LGTM with a comment

File: src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs (hunk 1 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs:349]: When reading the code from unsubscribe function, seems two events about select are misaligned, would you correct this case?

Comment (zeusongit) [reply src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs:349]: Updated

**Review by QilongTang — APPROVED** [review]

**Pull request merged**